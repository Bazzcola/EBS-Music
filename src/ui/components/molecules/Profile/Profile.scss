.profile {
    margin: 0 auto;
    max-width: 900px;
    display: flex;

    .profile_img {
        position: relative;
        width: 200px;
        height: 220px;
        border-radius: 10px;
        margin-right: 30px;

        img {
            width: 200px;
            height: 220px;
            border-radius: 10px;
        }

        button {
            position: absolute;
            bottom: 15px;
            left: 15px;
        }
    }
}

.profile_text {
    padding-top: 10px;

    span {
        padding: 3px;
        background: black;
        color: white;
        font-size: 10px;
        font-family: Open Sans, Arial, sans-serif;
    }

    h2 {
        padding-top: 10px;
        margin: 0px;
        font-family: Deezer, Arial, sans-serif;
        font-weight: 700;
    }

    p {
        font-weight: 400;
        font-family: Open Sans, Arial, sans-serif;
    }
}

.profile_circle_button {
    font-family: 'Nunito Sans', 'Helvetica Neue', Helvetica, Arial, sans-serif;
    font-weight: 700;
    border: 0;
    border-radius: 50%;
    cursor: pointer;
    display: inline-block;
    color: white;
    min-width: 40px;
    min-height: 40px;
    text-align: center;
    padding: 0px;
    background: grey;
}

.profile_circle_button_black {
    background-color: #000000 !important;
}

.profile_circle_button_white {
    background-color: #ffffff !important;
}

.profile_svg_play {
    display: flex;
    align-items: center;
    width: 15px;
    height: 15px;
    color: white;
    fill: currentColor;
    stroke: currentColor;
    stroke-width: 0;
    vertical-align: top;
    margin: 0 auto;
}

.profile_svg_play_black {
    color: black;
}

.profile_svg_stop {
    display: flex;
    align-items: center;
    width: 15px;
    height: 15px;
    color: white;
    fill: currentColor;
    stroke: currentColor;
    stroke-width: 0;
    vertical-align: top;
    margin: 0 auto;
}

.profile_svg_stop_black {
    color: black;
}

.profile_circle_button_hover {
    background-image: url("http://cdns-files.dzcdn.net/cache/slash/images/common/animated/equalizer/12x12/dark.ee54bef1c488b8caebde2b4915d0fc5d.gif");
    background-repeat: no-repeat;
    background-position: center;
}

.profile_circle_button_hover:hover {
    background-image: url("https://i.ibb.co/sbR3J4F/pausewhite.png");
    background-size: cover;
    z-index: 3;
    background-color: black;
}