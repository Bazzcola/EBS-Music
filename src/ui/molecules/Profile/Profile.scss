.profile {
    margin: 0 auto;
    max-width: 700px;
    display: flex;

    .profile_img {
        position: relative;
        width: 200px;
        height: 220px;
        border-radius: 10px;
        margin-right: 30px;

        button {
            position: absolute;
            bottom: 15px;
            left: 15px;
        }
    }
}

.profile_text {
    padding-top: 10px;

    span {
        padding: 3px;
        background: black;
        color: white;
        font-size: 10px;
        font-family: Open Sans, Arial, sans-serif;
    }

    h2 {
        padding-top: 10px;
        margin: 0px;
        font-family: Deezer, Arial, sans-serif;
        font-weight: 700;
    }

    p {
        font-weight: 400;
        font-family: Open Sans, Arial, sans-serif;
    }
}