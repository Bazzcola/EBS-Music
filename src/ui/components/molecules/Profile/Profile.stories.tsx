import React from 'react';
import { Profile } from 'ui/components/molecules/Profile/Profile';
import { MemoryRouter } from 'react-router-dom';

export default {
  title: 'Molecules/Profile',
  component: Profile
};

export const Profile1 = () => {
  return (
    <MemoryRouter>
      <Profile />
    </MemoryRouter>
  );
};
