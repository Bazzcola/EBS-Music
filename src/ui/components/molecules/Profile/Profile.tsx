import React, { useEffect, useState, useContext } from 'react';
import { useParams } from 'react-router-dom';
import { AudioContext, Tracks } from 'ui/context/audioContext';
import 'ui/components/molecules/Profile/Profile.scss';

type song = {
  id: number;
  title: string;
  name: string;
  src: string;
  duration: string;
  img: string;
};

export const Profile = () => {
  const black = 'black';
  const white = 'white';
  const match = useParams<{ name: string }>();
  const [track, setTrack] = useState<song>();
  const [active, setActive] = useState<boolean>(false);
  const { playing, setPlaying, setAudioFiles } = useContext(AudioContext);

  const changes = () => {
    setActive(!active);
    if (!active) {
      setPlaying(true);
    } else {
      setPlaying(false);
    }
  };

  useEffect(() => {
    if (playing) {
      setActive(true);
    }
    if (playing === false) {
      setActive(false);
    }
  }, [playing]);

  useEffect(() => {
    Tracks.filter((item: song) =>
      item.title === match.name ? setTrack(item) : false
    );
    Tracks.filter((item: song) =>
      item.title === match.name ? setAudioFiles(item.src) : false
    );
    // Tracks.filter((item: song) => (item.src ? setAudioFiles(item.src) : false));
    // eslint-disable-next-line
  }, [match]);

  return (
    <div className="profile">
      <div className="profile_img">
        <img src={track?.img} alt="" />
        {active ? (
          <button
            onClick={changes}
            className={`profile_circle_button profile_circle_button_white profile_circle_button_hover`}
          ></button>
        ) : (
          <button
            onClick={changes}
            className={`profile_circle_button profile_circle_button_white`}
          >
            <svg
              className={`profile_svg_stop profile_svg_stop_black`}
              viewBox="0 0 12 12"
            >
              <path d="M2.5.5v11l9-5.5z"></path>
            </svg>
          </button>
        )}
      </div>
      <div className="profile_text">
        <span>EXPLICIT</span>
        <h2>{track?.title}</h2>
        <p>{track?.name}</p>
      </div>
    </div>
  );
};
