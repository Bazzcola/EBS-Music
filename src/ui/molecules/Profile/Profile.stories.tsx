import React from 'react';
import { Profile } from 'ui/molecules/Profile/Profile';

export default {
  title: 'Molecules/Profile',
  component: Profile
};

export const Profile1 = () => {
  return <Profile />;
};
