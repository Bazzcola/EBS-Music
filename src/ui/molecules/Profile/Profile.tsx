import React from 'react';
import { CircleButton } from 'ui/atoms/CircleButton/CircleButton';
import 'ui/molecules/Profile/Profile.scss';

export const Profile = () => {
  const black = 'black';
  const white = 'white';

  return (
    <div className="profile">
      <div
        className="profile_img"
        style={{
          backgroundImage:
            'url(' +
            'https://akket.com/wp-content/uploads/2019/11/World-of-Warcraft-Shadowlands-Blizzard.jpg' +
            ')',
          backgroundPosition: 'center',
          backgroundSize: 'cover',
          backgroundRepeat: 'no-repeat'
        }}
      >
        <CircleButton classType={black} type={white} />
      </div>
      <div className="profile_text">
        <span>EXPLICIT</span>
        <h2>DECADANCE 2</h2>
        <p>Kyok</p>
      </div>
    </div>
  );
};
