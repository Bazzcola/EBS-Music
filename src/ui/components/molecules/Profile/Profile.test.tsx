import React from 'react';
import { render } from '@testing-library/react';
import { Profile } from './Profile';
import { MemoryRouter } from 'react-router-dom';

it('renders Profile correctly', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Profile />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
