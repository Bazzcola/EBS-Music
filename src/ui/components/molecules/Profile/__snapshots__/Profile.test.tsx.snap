// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Profile correctly 1`] = `
<DocumentFragment>
  <div
    class="profile"
  >
    <div
      class="profile_img"
    >
      <img
        alt=""
        src="https://i.pinimg.com/564x/2e/9a/d8/2e9ad8621e03cd81e2d81fcdbe327ba6.jpg"
      />
      <button
        class="profile_circle_button profile_circle_button_white"
      >
        <svg
          class="profile_svg_stop profile_svg_stop_black"
          viewBox="0 0 12 12"
        >
          <path
            d="M2.5.5v11l9-5.5z"
          />
        </svg>
      </button>
    </div>
    <div
      class="profile_text"
    >
      <span>
        EXPLICIT
      </span>
      <h2>
        Storybook
      </h2>
      <p>
        Storybook
      </p>
    </div>
  </div>
</DocumentFragment>
`;
